wire-ios-sync-engine/Source/Synchronization/SyncAgent.swift

```diff
@@ -69,7 +69,7 @@ final class SyncAgent: NSObject, SyncAgentProtocol {
     private var incrementalSyncToken: IncrementalSync.Token?
     private var ongoingSyncTask: Task<Void, Never>?
     private let conversationUpdatesGenerator: ConversationUpdatesGeneratorProtocol
-
+    private let commitPendingProposalsGenerator: CommitPendingProposalsGeneratorProtocol
```

Comment:
@johnxnguyen we talked about this, here I need another Generator which will clutter the SyncAgent.

Two solutions:

- Wrap the two generators into a GeneratorDirectory like the strategies
- Delegate the syncAgent events and move out the generators out of the SyncAgent.